rom/dev/src/flow/cold_reset/fmc_alias.rs

```diff
@@ -313,6 +313,27 @@ impl FmcAliasLayer {
             manifest.fmc.load_addr,
             manifest.fmc.size
         );
+        cprintln!(
+            "[afmc] Loading Runtime at address 0x{:08x} len {}",
+            manifest.runtime.load_addr,
+            manifest.runtime.size
+        );
+
+        // Detect if FMC and RT address ranges overlap
+        // Load address of FMC <= End address of RT
+        // and End address of FMC >= Load address of RT
+        if manifest.fmc.load_addr <= manifest.runtime.load_addr + manifest.runtime.size
+            && manifest.fmc.load_addr + manifest.fmc.size >= manifest.runtime.load_addr
```

Comment:
again, this check is only valid if we always load FMC at a lower address than RT. I would suggest checking for overlap the other way too, in case RT is ever loaded lower than FMC,

Reply:
This is the standard overlap detection logic. Equivalent to:

```
fmc.start_addr <= runtime.end_add &&
fmc.end_addr >= runtime.start_addr
```

(although though I would argue for use of < rather than <=)